### ISSUE TITLE
Broken when DYLD_LIBRARY_PATH is empty

### ISSUE THREAD

**Issue by elbaro** [Manishearth/compiletest-rs#179]
1. `cargo test` in OSX complains about empty `DYLD_LIBRARY_PATH`.
```
---- compile_test stdout ----
thread 'compile_test' panicked at 'Cannot link to dependencies. Problem with env var 'DYLD_LIBRARY_PATH': NotPresent', /Users/user/cargo/registry/src/github.com-1ecc6299db9ec823/compiletest_rs-0.3.22/src/common.rs:249:13
note: Run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
```

2. Manually setting DYLD_LIBRARY_PATH to "" leads to another error
because rustc is given two consecutive "-L" "-L":

```
DYLD_LIBRARY_PATH= cargo test

---- [run-pass] run-pass/lifetime4.rs stdout ----

error: compilation failed!
status: exit code: 1
command: "rustc" "tests/run-pass/lifetime4.rs" "-L" "/var/folders/qn/z1949f553jv89j0hnd5qzn9h0000gn/T/" "--target=x86_64-apple-darwin" "--error-format" "json" "-C" "prefer-dynamic" "-o" "/var/folders/qn/z1949f553jv89j0hnd5qzn9h0000gn/T/lifetime4.stage-id" "-L" "-L" "/var/folders/qn/z1949f553jv89j0hnd5qzn9h0000gn/T/lifetime4.stage-id.aux"
stdout:
------------------------------------------

------------------------------------------
stderr:
------------------------------------------
{"message":"multiple input filenames provided (first two filenames are `tests/run-pass/lifetime4.rs` and `/var/folders/qn/z1949f553jv89j0hnd5qzn9h0000gn/T/lifetime4.stage-id.aux`)","code":null,"level":"error","spans":[],"children":[],"rendered":"error: multiple input filenames provided (first two filenames are `tests/run-pass/lifetime4.rs` and `/var/folders/qn/z1949f553jv89j0hnd5qzn9h0000gn/T/lifetime4.stage-id.aux`)\n\n"}

------------------------------------------
```

3. `DYLD_LIBRARY_PATH=dummy cargo test` complains that my lib crate is unresolved. I just migrated to 2018 Edition. `extern crate my_lib` didn't work. The snippet in #150 didn't work. Can `compiletest-rs` be used in 2018 crates?

**Comment by joshlf** [comment]
It looks like this is related to this issue: https://github.com/nasa/europa/issues/181. I don't understand compiletest's internals well, but it seems like it's relying on DYLD_LIBRARY_PATH, which isn't going to work on newer versions of macOS, and the code in question may need to be changed.

**Comment by aldanor** [comment]
Any ideas/workarounds re: this? Just hit this as well...

(Might also be somewhat related: #155)

**Comment by laumann** [comment]
Doesn't #235 fix this? Looks like #233 is/was a dup of this one.

**Comment by joshlf** [comment]
It might fix it once it makes its way to crates.io 😛

**Comment by theypsilon** [comment]
This might be a good reason for a patch release.

**Comment by Munksgaard** [comment]
Yes, sorry about the delay. There's now a version 0.7.0 on crates.io which includes the fix in #235
